ocaml/xapi/xapi_cmd_result.ml

```diff
@@ -33,7 +33,11 @@ let add ?(sep = ':') line results =
 
 let find key results = RMap.find_opt key results
 
-let of_output_opt ~sep ~key ~lines =
+let of_output ~sep ~key lines =
   String.split_on_char '\n' lines
   |> List.fold_left (fun results line -> add ~sep line results) empty
-  |> find key
+  |> RMap.find key
+
+let of_output_opt ~sep ~key ~lines =
```

Comment:
Usually this is written the other way around: write the `_opt` function using other `_opt` functions and then write the exception raising function using the `_opt` functions. This is because raising an exception is more time consuming than returning `None`. If you wrap the exception returning function with Some/None then you might be catching more exceptions than intended, but also spending more time because an exception is raised (including capturing of a stacktrace) only to be ignored.

Reply:
I am not sure the rationale is sound. An exception in OCaml is very fast because it skips over all frames on the stack to the frame where is is caught (unlike in C++, for example). A regular return (like `None`) typically involves pattern matching in every function/frame until where it is properly handled. But exceptions are not tracked by the type system, making them more prone to programmer error.